### ISSUE TITLE
Can't build and export project, error: "cannot find module for page ..." in every /pages route

### ISSUE THREAD

**Issue by arochadaniel** [vercel/next.js#13464]
# Bug report
Can't build and export Next.js project, it gives me 
```
Error occurred prerendering page "/teams/[teamId]/advise/definitions". Read more: https://err.sh/next.js/prerender-error
Error: Cannot find module for page: /teams/[teamId]/advise/definitions
``` 
on every /pages route

## Describe the bug
I recently migrated my React.js + Apollo project to Next.js + Apollo, I followed all the examples for Apollo and managed to get it running in development (next dev) perfectly with the withApollo HOC. But when I need to build & export the project with next build, it compiles the project fine, but when it comes to automatically optimizing pages, it fails on every route saying that it cannot find the module for that page:

```
Error occurred prerendering page "/teams/[teamId]/advise/definitions". Read more: https://err.sh/next.js/prerender-error
Error: Cannot find module for page: /teams/[teamId]/advise/definitions
```

Not only in that route, but in every route available. Is worth mentioning that my pages folder is inside the src folder, I already tried building with the pages folder in the root directory, but ended up with the same results. I also tried to put every withApollo({ ssr: false }) page with ssr: true to see if the project builds fine, and all the pages without withApollo() still fail, so I don't think the withApollo HOC is the problem. The project is also built with Material UI, so I had to make some changes in the _document.tsx file and _app.tsx

_document.tsx
```
import { ServerStyleSheets } from '@material-ui/core/styles';
import Document, { Head, Html, Main, NextScript } from 'next/document';
import React from 'react';

import { theme } from '../modules/core-modules/services/material-ui/theme';

export default class MyDocument extends Document {
  render() {
    return (
      <Html lang='en'>
        <Head>
          <meta name='theme-color' content={theme.palette.primary.main} />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

MyDocument.getInitialProps = async (ctx) => {
  const sheets = new ServerStyleSheets();
  const originalRenderPage = ctx.renderPage;

  ctx.renderPage = () =>
    originalRenderPage({
      enhanceApp: (App) => (props) => sheets.collect(<App {...props} />),
    });

  const initialProps = await Document.getInitialProps(ctx);

  return {
    ...initialProps,
    // Styles fragment is rendered after the app and page rendering finish.
    styles: [
      ...React.Children.toArray(initialProps.styles),
      sheets.getStyleElement(),
    ],
  };
};

```

_app.tsx
```
import './base-styles.css';

import { CssBaseline } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/styles';
import { useRouter } from 'next/router';
import React, { useEffect } from 'react';

import { isFreeTrialOver } from '../modules/core-modules/routing/free-trial-locker/is-free-trial-over';
import Locker from '../modules/core-modules/routing/free-trial-locker/locker';
import { theme } from '../modules/core-modules/services/material-ui/theme';
import getDecodedToken from '../modules/feature-modules/authentication/helpers/get-decoded-token';
import SeoHelmet from '../modules/feature-modules/shared/seo/helmet';

const MyApp = (props: {
  Component: any;
  pageProps: { [key: string]: any };
}) => {
  const { Component, pageProps } = props;
  const router = useRouter();

  const jwt = getDecodedToken('jwt');

  useEffect(() => {
    const jssStyles: any = document.querySelector('#jss-server-side');
    if (jssStyles !== null) {
      jssStyles.parentElement.removeChild(jssStyles);
    }

    if (!jwt) {
      if (
        router.pathname.match(/^\/enterprises.*/) ||
        router.pathname.match(/^\/teams.*/)
      ) {
        router.push('/');
      }
    } else if (jwt) {
      if (router.pathname.match(/^\/authentication.*/)) {
        router.push('/enterprises');
      }
    }
  }, []);

  const TrialIsOver = () => {
    if (jwt && jwt.type === 'free' && isFreeTrialOver(jwt)) {
      return <Locker />;
    }

    return <Component {...pageProps} />;
  };

  return (
    <ThemeProvider theme={theme}>
      <SeoHelmet />
      <CssBaseline />
      <TrialIsOver />
    </ThemeProvider>
  );
};

export default MyApp;

```

## To Reproduce

1. Migrate React.js project to Next.js
2. Run npm run build (next build)
3. Wait for compile success
4. See error on pages automatic optimization

## Expected behavior
It should compile and optimize with no errors 

## Screenshots

[https://imgur.com/AMP20ff](https://imgur.com/AMP20ff)

## System information

- OS: Xubuntu 20.04
- Version of Next.js: 9.4.2
- Version of Node.js: 10.19.0

**Comment by arochadaniel** [comment]
I fixed the issue, for anyone that could have the same problem. I was generating an ID for the build based on the last commit hash, I removed the generateBuildId property in my next.config.js and the problem was fixed, now I'm building and exporting without problems. I did that to keep an order and history between builds and git, but is causing me problems, maybe it is a bug?

See [the docs](https://nextjs.org/docs/api-reference/next.config.js/configuring-the-build-id) for more information on generateBuildId. 

Closing
